trie/extensionNode.go

```diff
@@ -268,48 +240,59 @@ func (en *extensionNode) getEncodedNode() ([]byte, error) {
 	return marshaledNode, nil
 }
 
-func (en *extensionNode) resolveCollapsed(_ byte, db common.TrieStorageInteractor) error {
-	err := en.isEmptyOrNil()
-	if err != nil {
-		return fmt.Errorf("resolveCollapsed error %w", err)
+func (en *extensionNode) isCollapsed() bool {
+	return en.child == nil && len(en.EncodedChild) != 0
+}
+
+func (en *extensionNode) resolveIfCollapsed(db common.TrieStorageInteractor) (node, error) {
+	en.childMutex.Lock()
+	defer en.childMutex.Unlock()
+
+	isChildCollapsed := en.child == nil && len(en.EncodedChild) != 0
+	if !isChildCollapsed {
```

Comment:
why not use en.isCollaped? exactly the same functions is wrote on top of this function.

Reply:
Changed.